cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -555,13 +554,15 @@ protected StreamSizeAggregator(Id.Stream streamId, long baseCount, int streamThr
       this.streamInitSize = baseCount;
       this.streamThresholdMB = new AtomicInteger(streamThresholdMB);
       this.cancellable = cancellable;
-      this.isInit = true;
       this.streamId = streamId;
       this.streamFeed = new Id.NotificationFeed.Builder()
         .setNamespaceId(streamId.getNamespaceId())
         .setCategory(Constants.Notification.Stream.STREAM_FEED_CATEGORY)
         .setName(String.format("%sSize", streamId.getName()))
         .build();
+
+      // Publish an initialization notification
+      publishNotification(baseCount);
```

Comment:
With notifications we are okay if a publish fail. I'll move the publish call, like you suggested.